src/loaders/utils/GetTakeoverDetails.ts

```diff
@@ -0,0 +1,96 @@
+import { isServerSide } from 'utils/getEnv';
+import { getArticleVideoServiceBaseUrl } from 'utils/getPandoraDetails';
+import { trackYoutubeTakeoverDetails, YoutubePlayerTrackingProps } from 'youtube/players/shared/youtubeTrackingEvents';
+import { YoutubeTakeOverDetails } from 'youtube/types';
+import { getMediaWikiConfigDetails } from 'loaders/utils/GetMediaWikiConfigDetails';
+import { VimeoTakeOverDetails } from 'vimeo/types';
+
+function getTakeoverUrl(wikiId: string): string {
+	const articleVideoBaseUrl = getArticleVideoServiceBaseUrl();
+	if (articleVideoBaseUrl.length === 0) {
+		return articleVideoBaseUrl;
+	}
+	if (!wikiId) {
+		return '';
+	}
+	return `${articleVideoBaseUrl}takeover/v1/takeover-mappings/${wikiId}`;
+}
+
+export async function getTakeoverDetails({ deviceType }: YoutubePlayerTrackingProps) {
+	const { wikiId, isTier3Wiki } = getMediaWikiConfigDetails();
+	if (!wikiId || isTier3Wiki) {
+		return null;
+	}
+
+	const response = await fetch(getTakeoverUrl(wikiId));
+	const dataArray = await response.json();
+	if (dataArray.length === 0) {
```

Comment:
```const [data] = await response.json();
if (!data) {
```

Reply:
But will this guard from an empty array?